osgp/protocol-adapter-dlms/osgp-jasper-interface/src/main/java/org/opensmartgridplatform/adapter/protocol/jasper/rest/client/JasperWirelessTerminalRestClient.java

```diff
@@ -60,11 +60,34 @@ public GetSessionInfoResponse getSession(final String iccId) throws OsgpJasperEx
 
       getSessionInfoResponse = getSessionInfoResponseEntity.getBody();
 
+      getSessionInfoResponse = this.checkOnSessionValidity(getSessionInfoResponse);
+
```

Comment:
Now the same variable is used twice, would it not be nice to make a method:
sessionInfoResponse = getValidSessionInfoResponse(getSessionInfoResponseEntity)